evento/src/views/App/index.js

```diff
@@ -9,25 +9,83 @@ import RegisterPage from '../RegisterPage';
 import PathNotFound from './components/PathNotFound';
 import './App.css';
 
+import api from '../../api';
+import session from '../../session';
+
+const AuthStatus = {
+	Authenticating: 0,
+	Authenticated: 1,
+	NotAuthenticated: 2,
+};
+
 class App extends Component {
+	constructor() {
+		super();
+		this.state = {
+			authenticating: true,
+			errorMessage: null,
+		};
+	}
+
+	async componentDidMount() {
+		this.authenticate();
+	}
 
 	render() {
 		return (
 			<div className="App">
 				<Router basename="/evento">
-					<Switch>
-						<Route exact path='/' component={MainPage} />
-						<Route exact path='/events' component={MainPage} />
-						<Route exact path='/signin' component={SignInPage} />
-						<Route exact path='/register' component={RegisterPage} />
-						<Route exact path='/event/new' component={NewEventPage} />
-						<Route exact path='/event/:eventId' component={EventPage} />
-						<Route path='*' component={PathNotFound} />
-					</Switch>
+					{this.getContent()}
 				</Router>
 			</div>
 		);
 	}
+
+	getContent() {
+		if(this.state.errorMessage) {
+			return <h4>{ this.state.errorMessage }</h4>;
+		}
+
+		const authStatus = this.getAuthStatus();
+		if(authStatus === AuthStatus.Authenticating) {
+			return <h4>loading...</h4>;
+		}
+		else if(authStatus === AuthStatus.NotAuthenticated) {
+			return <Route component={SignInPage} />;
```

Comment:
One problem with this atm: RegisterPage and SignInPage are different components, and thus when user on the sign in page clicks "Not yet registered?" link, the site doesn't move to RegisterPage component (because this line forces it to be SignInPage always if the auth status == NotAuthenticated).

I think we should just combine SignInPage and RegisterPage into a single URL.